core/optaplanner-core-impl/src/main/java/org/optaplanner/core/config/constructionheuristic/decider/forager/ConstructionHeuristicForagerConfig.java

```diff
@@ -22,6 +22,15 @@ public void setPickEarlyType(ConstructionHeuristicPickEarlyType pickEarlyType) {
         this.pickEarlyType = pickEarlyType;
     }
 
+    // ************************************************************************
+    // With methods
+    // ************************************************************************
+
+    public ConstructionHeuristicForagerConfig withPickEarlyType(ConstructionHeuristicPickEarlyType pickEarlyType) {
+        this.setPickEarlyType(pickEarlyType);
+        return this;
```

Comment:
If we need `self`, let's at least use it consistently.
(I'm not entirely convinced we need `self`.)

Reply:
The `self()` method is needed only on the abstract config classes; everywhere else `this` would do. I see the `self()` as a minor complication, so I was hesitant about putting it in every config class. On the other hand, I agree that consistency is always welcome.